### ISSUE TITLE
Support "signcryption"

### ISSUE THREAD

**Issue by ribose-jeffreylau** [riboseinc/transcryptor#9]
Roadmap #3

**Comment by GildedHonour** [comment]
I'll work on this.

**Comment by GildedHonour** [comment]
1) Signcryption requires an assymetric algorithm, doesn't it? Whereas now transcryptor, at least by default, uses AES which is symmetric one. Should we change that so that it supports only assymetric algorithms?

2) Will signcryption change the interface for a client? Or will a client use transcryptor the way he uses it now?

**Comment by ronaldtse** [comment]
@GildedHonour indeed. The current usage of transcryptor is locked into attr-encrypted which relies on the encryptor gem, which in turn relies on OpenSSL.

To enable signcryption we will need to support algorithms beyond those provided by the encryptor gem, that uses non-OpenSSL libraries, such as https://github.com/randombit/botan, for example through https://github.com/riboseinc/ruby-botan.

Signcryption algorithms are also detailed in [ISO/IEC 29150:2011 -- Information technology -- Security techniques — Signcryption](https://www.iso.org/standard/45173.html).

**Comment by GildedHonour** [comment]
@ronaldtse ISO/IEC 29150:2011 isn't free and I haven't found a free one in the internet, is there any? Although I have a few pdfs and code in python, maybe that'll be enough.
